### ISSUE TITLE
can't install on windows

### ISSUE THREAD

**Issue by amitport** [flutter/flutter#14884]
@goderbauer

## Steps to Reproduce

follow the windows instructions:

- git clone dev branch (https://github.com/flutter/flutter/commit/3ea4d06340a97a1e9d7cae97567c64e0569dcaa2)
- add to path 
- run `flutter doctor`

## Logs

flutter doctor output:
```
Updating flutter tool...
The system cannot find the path specified.
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds...

Waiting for 0 seconds, press CTRL+C to quit ...
```

## Flutter Doctor

does not run

**Comment by aam** [comment]
Thanks for the report. Would you be able to edit `bin\flutter.bat`: replace first line `@ECHO off` with `@ECHO on`, rerun `flutter doctor` and paste the output? This should help with troubleshooting the problem.

**Comment by tvolkert** [comment]
@amitport can you also see if this happens if you `git reset --hard f914e70`?  I'm interested to see if our v0.1.4 release is affected.

**Comment by tvolkert** [comment]
Removing TODAY label since I've been unable to repro on a local Windows machine.

**Comment by Panthro** [comment]
I can confirm the issue with a fresh install on a windows 8.1.

```
C:\dev\flutter\packages\flutter_tools>SET PUB_ENVIRONMENT=:flutter_install

C:\dev\flutter\packages\flutter_tools>IF "" == "" (IF EXIST "C:\dev\flutter\.pub-cache" SET PUB_CACHE=C:\dev\flutter\.pub-cache )

C:\dev\flutter\packages\flutter_tools>CALL "C:\dev\flutter\bin\cache\dart-sdk\bin\pub.bat" upgrade --verbosity=error --no-packages-dir
The system cannot execute the specified program.
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds...

Waiting for 0 seconds, press CTRL+C to quit ...
The system cannot execute the specified program.
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds...

Waiting for 0 seconds, press CTRL+C to quit ...
The system cannot execute the specified program.
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds...

Waiting for 3 seconds, press CTRL+C to quit ...
Terminate batch job (Y/N)?
```

**Comment by Panthro** [comment]
@tvolkert also doing a `git reset --hard f914e70` on a freshly cloned before doing the doctor didn't help

**Comment by Panthro** [comment]
`echo on` on `pub.bat` gives me the following:

```
C:\dev\flutter\bin\cache\dart-sdk>rem We allow extra vm options to be passed in through an environment variable.

C:\dev\flutter\bin\cache\dart-sdk>if not "__" == "__" (set VM_OPTIONS=  )

C:\dev\flutter\bin\cache\dart-sdk>"C:\dev\flutter\bin\cache\dart-sdk\bin\dart"  "C:\dev\flutter\bin\cache\dart-sdk\bin\snapshots\pub.dart.snapshot" up
grade
The system cannot execute the specified program.

C:\dev\flutter\bin\cache\dart-sdk>endlocal
```

**Comment by Panthro** [comment]
I can confirm it was the McAfee antivirus that was blocking the execution, moved the whole flutter folder to an AV ignored folder and the upgrade worked

**Comment by Panthro** [comment]
Then it failed because the ignored folder contains a ` ` in it.

https://github.com/flutter/flutter/pull/14955

**Comment by goderbauer** [comment]
@Panthro What was the failure message that you saw when you had a space in the path?

**Comment by goderbauer** [comment]
@amitport what does the folder structure inside the `cache` directory in the flutter dir look like? Can you also try cloning the `master` branch and see if you still have the issue with that one? @aam recently fixed a path handling issue that might be the cause for your problem.

**Comment by Panthro** [comment]
@goderbauer `Microsoft is not expected here` or something like that, because my ignored path is a `Microsoft something` inside `Program Files (x86)`

Sorry, but today I don't have spare time to reproduce again, I was in the `beta` branch, which might be broken as per #14955 comments

**Comment by zoechi** [comment]
dup of #15014 ?

**Comment by zoechi** [comment]
@Panthro is this still an issue?
Spaces in project and sdk should be fixed dart-lang/pub#1882 (comment)

**Comment by zoechi** [comment]
This issue seems to be fixed.
